compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNOps.td

```diff
@@ -191,4 +195,101 @@ def CUDNN_GetCurrentHandle : CUDNN_Op<"get_current_handle", [Pure]> {
     let assemblyFormat = [{ attr-dict type($handle) }];
 }
 
+//===----------------------------------------------------------------------===//
+// cudnn.graph operation
+//===----------------------------------------------------------------------===//
+
+def CUDNN_GraphOp : CUDNN_Op<"graph", [
+    IsolatedFromAbove,
+    FunctionOpInterface,
+    CallableOpInterface,
+    Symbol
+  ]> {
+  let summary = "operation defining cuDNN graph building function";
+
+  let description = [{
+    Represents a function building a cuDNN graph from a cuDNN operations. At
+    run time this operation lowered to vm function calling into cuDNN module API
+    to construct a cuDNN operation graph object.
+  }];
+
+  let arguments = (ins
+    TypeAttrOf<FunctionType>:$function_type,
+    OptionalAttr<DictArrayAttr>:$arg_attrs,
+    OptionalAttr<DictArrayAttr>:$res_attrs
+  );
+
+  let regions = (region AnyRegion:$body);
+
+  let extraClassDeclaration = [{
+    /// Returns the argument types of cuDNN graph.
+    llvm::ArrayRef<mlir::Type> getArgumentTypes() {
```

Comment:
Might be worth casting to cudnn tensor type?

Reply:
I think it will also have `IntegerType` (or index) once dynamic shape support will land